homeassistant/components/lyric/sensor.py

```diff
@@ -184,6 +194,44 @@ def get_setpoint_status(status: str, time: str) -> str | None:
                         )
                     )
 
+            if device.macID in coordinator.data.rooms_dict:
+                room: LyricRoom
+                for room in coordinator.data.rooms_dict[device.macID].values():
+                    if hasattr(room, "roomAvgTemp"):
+                        entities.append(
+                            LyricRoomSensor(
+                                coordinator,
+                                LyricRoomSensorEntityDescription(
+                                    key=f"{device.macID}_room{room.id}_temperature",
+                                    name=f"{room.roomName} Average Temperature",
+                                    device_class=SensorDeviceClass.TEMPERATURE,
+                                    state_class=SensorStateClass.MEASUREMENT,
+                                    native_unit_of_measurement=hass.config.units.temperature_unit,
```

Comment:
This should return the unit of measurement from the device, not from hass itself.